lib/prediction/index.js

```diff
@@ -0,0 +1,270 @@
+ *
+ * @constructor
+ * @alias module:prediction
+ *
+ * @param {object} options - [Configuration object](#/docs/?method=gcloud).
+ *
+ * @example
+ * var gcloud = require('gcloud')({
+ *   keyFilename: '/path/to/keyfile.json',
+ *   projectId: 'grape-spaceship-123'
+ * });
+ *
+ * var prediction = gcloud.prediction();
+ */
+function Prediction(options) {
+  if (!(this instanceof Prediction)) {
+    options = util.normalizeArguments(this, options);
+    return new Prediction(options);
+  }
+
+  var config = {
+    baseUrl: 'https://www.googleapis.com/prediction/v1.6',
+    scopes: [
+      'https://www.googleapis.com/auth/prediction',
+      'https://www.googleapis.com/auth/devstorage.read_only'
+    ]
+  };
+
+  Service.call(this, config, options);
+}
+
+nodeutil.inherits(Prediction, Service);
+
+/**
+ * Create a trained model.
+ *
+ * You may optionally provide a {module:storage/file} as a data source to train
+ * the new model. If you have a CSV file, but haven't put it in a bucket yet,
+ * you will need to reference a bucket with {module:storage/bucket}, a file with
+ * {module:storage/file}, and upload it with {module:storage/file#upload}.
+ *
+ * @resource [Trainedmodels: insert API Documentation]{@link https://cloud.google.com/prediction/docs/reference/v1.6/trainedmodels/insert}
+ *
+ * @throws {error} If a model ID is not provided.
+ *
+ * @param {string} id - Unique name for the model. E.g. "my-model"
+ * @param {object=} options - Configuration object.
+ * @param {module:storage/file} options.data - A CSV file to use as training
+ *     data.
+ * @param {string} options.type - The type of model, `classification`
+ *     (categorical, text labels) or `regression` (numeric labels). Learn more
+ *     about these options in the [Developers Guide](https://cloud.google.com/prediction/docs/developer-guide#examples).
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An API error.
+ * @param {?module:prediction/model} callback.model - A new
+ *     {module:prediction/model} object.
+ * @param {object} callback.apiResponse - Raw API response.
+ *
+ * @example
+ * prediction.createModel('my-model', function(err, model, apiResponse) {
+ *   // `model` is a Model object.
+ * });
+ *
+ * //-
+ * // Create a model, using an existing CSV file in a Google Cloud Storage
+ * // bucket as training data.
+ * //-
+ * var gcs = gcloud.storage();
+ * var bucket = gcs.bucket('my-bucket');
+ * var modelDataCsv = bucket.file('my-model.csv');
+ *
+ * prediction.createModel('my-model', {
+ *   data: modelDataCsv
+ * }, function(err, model, apiResponse) {});
+ */
+Prediction.prototype.createModel = function(id, options, callback) {
+  var self = this;
+
+  if (!id) {
+    throw new Error('A model ID is required.');
+  }
+
+  if (is.fn(options)) {
+    callback = options;
+    options = {};
+  }
+
+  var body = extend({}, options, {
+    id: id
+  });
+
+  var file = body.data;
```

Comment:
should we be deleting `body.data`? I didn't see it in the upstream request body docs.